### PR TITLE
Fix: Value Type of Identification in API Response

### LINKED ISSUE #58: Value Type of Identification in API Response
## Issue

Based on the document from jsonapi.org:
```
Every resource object MUST contain an id member and a type member. The values of the id and type members MUST be strings.
```

Here is the link: https://jsonapi.org/format/#document-resource-object-identification

Currently, the value type of `id` is `Int`

Screen Shot 2021-06-17 at 19 04 29

## Expected

The value type of `id` should be `String

## Steps to reproduce

1. Login with API `{{base_url}}/api/v1/oauth/token`
2. Check the response

### PR DESCRIPTION
**Pull request by hoangmirs**
Resolve #58 

## What happened
Fix: Value Type of Identification in API Response

 
## Insight
Change the custom response of Doorkeeper token API

## Proof Of Work
The response would be like this:
```json
{
    "data": {
        "id": "5",
        "type": "token",
        "attributes": {
            "access_token": "93N1shlgeZn4fFwmYiSMWPNV-42WuhA5vuPMmIem2Ik",
            "token_type": "Bearer",
            "expires_in": 7200,
            "refresh_token": "iJ3G_pKCJioyDw22zfFBYp2ExVRd8Vt0lGtKNgjNWvM",
            "created_at": 1623987725
        }
    }
}
```

### CODE REVIEW

File: spec/controllers/api/v1/tokens_controller_spec.rb [hunk 1]
```diff
@@ -3,6 +3,113 @@
 require 'rails_helper'
 
 RSpec.describe API::V1::TokensController, type: :controller do
+  describe 'POST#create' do
+    context 'given a valid oauth application' do
+      context 'given valid user credentials' do
+        it 'returns success status' do
+          application = Fabricate(:application)
+          user = Fabricate(:user, email: 'dev@nimblehq.co', password: '12345678')
```
Comment (team-nimblehq) [inline spec/controllers/api/v1/tokens_controller_spec.rb:11]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/29430cc9172c80e61e2206f65e8842ea68b7ce76/spec/controllers/api/v1/tokens_controller_spec.rb#L11"></span>Useless assignment to variable - <code>user</code>.</td>
    </tr>
  </tbody>
</table>

File: spec/controllers/api/v1/tokens_controller_spec.rb [hunk 2]
```diff
@@ -3,6 +3,113 @@
...
+          }.merge(oauth_application_params(application))
+
+          post :create, params: params
+
+          expect(response).to have_http_status(:success)
+        end
+
+        it 'returns an empty response' do
+          application = Fabricate(:application)
+          user = Fabricate(:user, email: 'dev@nimblehq.co', password: '12345678')
```
Comment (team-nimblehq) [inline spec/controllers/api/v1/tokens_controller_spec.rb:26]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/29430cc9172c80e61e2206f65e8842ea68b7ce76/spec/controllers/api/v1/tokens_controller_spec.rb#L26"></span>Useless assignment to variable - <code>user</code>.</td>
    </tr>
  </tbody>
</table>

File: spec/controllers/api/v1/tokens_controller_spec.rb [hunk 3]
```diff
@@ -3,6 +3,113 @@
...
+
+          response_body = JSON.parse(response.body)
+          expect(response_body).to match_json_schema('v1/tokens/create/invalid')
+        end
+      end
+    end
+
+    context 'given an invalid oauth application' do
+      it 'returns unauthorized status' do
+        user = Fabricate(:user, email: 'dev@nimblehq.co', password: '12345678')
```
Comment (team-nimblehq) [inline spec/controllers/api/v1/tokens_controller_spec.rb:75]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/29430cc9172c80e61e2206f65e8842ea68b7ce76/spec/controllers/api/v1/tokens_controller_spec.rb#L75"></span>Useless assignment to variable - <code>user</code>.</td>
    </tr>
  </tbody>
</table>

File: spec/controllers/api/v1/tokens_controller_spec.rb [hunk 4]
```diff
@@ -3,6 +3,113 @@
...
+          password: '12345678'
+        }
+
+        post :create, params: params
+
+        expect(response).to have_http_status(:unauthorized)
+      end
+
+      it 'returns an error message' do
+        user = Fabricate(:user, email: 'dev@nimblehq.co', password: '12345678')
```
Comment (team-nimblehq) [inline spec/controllers/api/v1/tokens_controller_spec.rb:89]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/29430cc9172c80e61e2206f65e8842ea68b7ce76/spec/controllers/api/v1/tokens_controller_spec.rb#L89"></span>Useless assignment to variable - <code>user</code>.</td>
    </tr>
  </tbody>
</table>

**Comment by team-nimblehq** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/eb6086d32e72ab5ceb99893a972dc2d3ef617cc9/lib/doorkeeper/custom_error_response.rb#L5">lib/doorkeeper/custom_error_response.rb#L5</a> - Method has too many lines. [11/10]</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/models/survey.rb [hunk 5]
```diff
@@ -29,9 +29,7 @@ class Survey < ApplicationModel
 
   class << self
     def all
-      @all ||= begin
-        JSON.parse(File.read(Rails.root.join('lib', 'assets', 'surveys.json'))).map(&method(:new))
-      end
+      @all ||= JSON.parse(File.read(Rails.root.join('lib', 'assets', 'surveys.json'))).map(&method(:new))
     end
 
     def where(attributes = {})
```
Comment (team-nimblehq) [inline app/models/survey.rb:35]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/d2ead8d9256331d362ed7c6b15d34734392b6c41/app/models/survey.rb#L35"></span>Remove unused methods (survey#where)</td>
    </tr>
  </tbody>
</table>

**Review by junan — APPROVED** [review]

**Review by ankitkalia1195 — APPROVED** [review]

**Pull request merged**